server/middleware/ensureActiveCaseLoadSet.ts

```diff
@@ -0,0 +1,43 @@
+import { RequestHandler } from 'express'
+import logger from '../../logger'
+import UserService from '../services/userService'
+import { CaseLoad } from '../data/interfaces/caseLoad'
+import { PrisonUser } from '../interfaces/prisonUser'
+
+export function ensureActiveCaseLoadSet(userService: UserService): RequestHandler {
+  return async (req, res, next) => {
+    try {
+      if (res.locals.user && res.locals.user.activeCaseLoad && res.locals.user.activeCaseLoadId) return next()
+
+      if (res.locals.user && res.locals.user.caseLoads) {
+        const activeCaseLoad = await getActiveCaseload(res.locals.user.caseLoads, userService, res.locals.user)
+        res.locals.user.activeCaseLoad = activeCaseLoad
+        res.locals.user.activeCaseLoadId = activeCaseLoad?.caseLoadId
+      }
+      return next()
+    } catch (error) {
+      logger.error(error, `Failed to initialise active case load for: ${res.locals.user && res.locals.user.username}`)
+      return next(error)
```

Comment:
Should be we erroring at this point? We might actually have a potential caseload for this interaction, even though it's not been saved to NOMIS

Reply:
As discussed, we do want to stop users from navigating away from the homepage without having an active case load set